### DISCUSSION TITLE
ESPHome / bluetooth proxy?

### DISCUSSION THREAD

**Discussion by rovingclimber** [fl4p/batmon-ha#27] [Ideas]
Hi!
This project looks really interesting ... I'm wondering if it's possible to use this add-on via ESPHome BLE proxy? That would extend the range out beyond BLE range (and beyone ... since you can get ESPHome to connect via Wireguard) ... would be great for remote monitoring your BMS.

**Comment by fl4p** [comment]
Hi,
did you try https://github.com/syssi/esphome-jbd-bms ? This is a ESPhome add-on for varios BMS

**Reply by edwardkusz** [reply]
This is how to use it via uart  but maybe there is a possibility to use esp32bluetooth proxy and read directly from bms bluetooth and send to batmon ha

**Reply by JochenGillich** [reply]
I also want to use batmon over the esphome bluetooth proxy. But it did not work. It works only over a direkt bluetooth connection.

**Reply by mobilex-Godelmann** [reply]
any update in this regard? My plan was to plant an esp next to the battery. Don't want to mess with opening the casing...
Also using a full raspy seem a little overkill.

**Reply by syssi** [reply]
The mentioned project above (esphome-jbd-bms) supports BLE too.

**Reply by mobilex-Godelmann** [reply]
I'll take a look again. I saw your project, but I thought it required to be connected direclty to UART connector inside the battery.
My battery comes equipped with bluetooth module.

**Reply by syssi** [reply]
Both transports (UART & BLE) are supported.

**Reply by fl4p** [reply]
I tested batmon on a Raspberry Pi. Since it uses bleak for communication with bluetooth stack, it'll run on any Windows, Linux, macOs or Android platform.
Instead of an ESP32 you can use a Raspberry PI Zero or similar.

**Reply by fl4p** [reply]
Just as note,  use MicroPython to run batmon on esp32 and use MicroPython's [bluetooth module](https://docs.micropython.org/en/latest/library/bluetooth.html) (maybe port bleak)

**Comment by rovingclimber** [comment]
Ah, great, thanks!

**Comment by Raydox187** [comment]
hey, this is my first HASS project and i test it some days with normal bluetooth and 4 xBMS. but the connection stops and lost everytime after some seconds with my daly bms. i changed much but not working. now i buy atom lite and hope it works like bluetooth dongle but i do not know how change adapter for this in batmon or is it not possible . i am a absolut newbie.
how can i impliment Proxy in Adapter ? or is it not possible? adapter: "atomESP or MAC ?

thanks for help"

**Reply by fl4p** [reply]
I'd suggest to run batmon outside of  HA and docker. See doc/Standalone.md

**Comment by Battlewind** [comment]
Adding the Bluetooth proxy to this would be awesome!  I already have Bluetooth modules for the two BMS I have. Also have the USB modules.  Not much into wiring the ESP32 to each BMS.

**Reply by wurmtal868** [reply]
I double down on this request. 
The "esphome-jbd-bms" project does not support my Daly 8S/50A BMS. There is already a Bluetooth proxy near my battery for a room temperature sensor. So it would be nice if the batmon could make use of this proxy.

**Comment by zendick** [comment]
Please add BLE proxy support

**Comment by Tomblarom** [comment]
Is there a chance to add BLE proxy support? 😫 I have multiple BT proxies installed, in form of Shellys and dedicated [ESP32 Bluetooth Proxy](https://esphome.io/projects/index.html).. I have them, because I use multiple [Shelly BLU Wall Switch 4](https://www.shelly.com/de/products/shelly-blu-wall-switch-4-stand-alone-test-bundle), so it would be sooo convenient, to use them all together, to connect to my JBD batteries..! 😊

**Reply by syssi** [reply]
Do you know this alternative? https://github.com/patman15/BMS_BLE-HA

**Reply by Tomblarom** [reply]
@syssi I just got started testing :) Will give it a try. Thanks!

**Reply by syssi** [reply]
Patricks implementation supports [ESPHome Bluetooth proxy](https://esphome.io/components/bluetooth_proxy) for sure!

**Reply by FrankieHAi** [reply]
I prefer the functionality of your implementation, bluetooth proxy support would be very useful

**Reply by saviothecnic** [reply]
I use BLE Battery Management System
for Seplos BMS V 2.0 and it works fine
using EspHome as GW BT
while BatMon doesn't seem to work with EspHome BT proxy.

I'd be happy if you could add this feature
because BatMon reads my daily data very well and has many sensors
plus MQTT, which BLE Battery Management System doesn't do.
So I'd be happy if BatMon
added support for the EspHome GW BT
and obviously if in the future it also supports BT reading for
Seplos BMS V 2.0 so I only use one integration, even two.
